cscore/src/main/java/edu/wpi/first/cscore/CvSink.java

```diff
@@ -71,7 +71,25 @@ public CvSink(String name) {
    * @return Frame time, or 0 on error (call GetError() to obtain the error message)
    */
   public long grabFrame(Mat image) {
-    return grabFrame(image, 0.225);
+    return grabFrame(image, 0.225, 0);
+  }
+
+  /**
+   * Wait for the next frame and get the image. Times out (returning 0) after 0.225 seconds. The
+   * provided image will have the pixelFormat this class was constructed with.
+   *
+   * <p>If lastFrameTime is provided and non-zero, the sink will fill image with the first frame
+   * from the source that is not equal to lastFrameTime. If lastFrameTime is zero, the time of the
+   * current frame owned by the CvSource is used, and this function will block until the connected
+   * CvSource provides a new frame.
+   *
+   * @param image Where to store the image.
+   * @param lastFrameTime Timestamp of the last frame - used to compare new frames against, or 0 to
+   *     use the current time
+   * @return Frame time, or 0 on error (call GetError() to obtain the error message)
+   */
+  public long grabFrame(Mat image, long lastFrameTime) {
```

Comment:
Because of the way this creates potentially confusing overloads (passing long vs double has *quite* different meaning here), I think we should name the lastFrameTime functions differently rather than overloading.

Reply:
I can do this pretty easily in Java land - C++ might be harder without introducing some code duplication. Do we want to expose a normal and a last frame time version of every function, or only the ones I actually need?

Reply:
Maybe just the ones you need for now?  This probably needs a bigger refactor to avoid duplication than we want to do right now.

Reply:
Sounds good, I can make that happen. It'd also be nice to think about how we can avoid having to go from a native RawFrame -> NewDirectByteBuffer -> CvMat from the bytebuffer -> copy the mat into the user-provided one if we're already providing a parallel API, too

Reply:
We could give the native function a native pointer, size, stride, etc, to copy into, but the problem is how to handle the case when that doesn’t match the image size it has available, and it still makes a copy of course.

The other approach would be allowing zero copy by returning an object pointing to the internal RawFrame native buffer, and calling close() on that object returns the native buffer to the pool.  This puts the onus on the user to do so or have a memory leak.

Reply:
Updated the the API to expose a new function for what we need, leaving everything else untouched.